CAIPs/caip-207.md

```diff
@@ -0,0 +1,236 @@
+---
+caip: 207
+title: JSON-RPC Authority Negotiation
+author: Pedro Gomes (@pedrouid), Hassan Malik (@hmalik88)
+discussions-to: https://github.com/ChainAgnostic/CAIPs/pull/207
+status: Draft
+type: Standard
+created: 2023-02-02
+updated: 2023-02-02
+requires: [2, 10, 25, 171]
+---
+
+## Simple Summary
+
+CAIP-207 extends CAIP-25 to allow callers and respondents to negotiate RPC
+authorities and RPC semantics before authorizing RPC terms. 
+
+## Abstract
+
+CAIP-207 defines additional properties that CAIP-25 can be used
+iteratively/progressively to layer custom or local RPC semantics and/or routing
+onto a session. Since CAIP-25 respondents ignore unknown properties, respondents
+that conform to CAIP-25 but not to CAIP-207 should be carefully considered and
+accomodated by implementers. 
+
+## Motivation
+
+While some core methods and notifications are foundational to a namespace and
+universally defined out-of-band (meaning all callers and respondents agree to
+them already), others are specific to chains or even to subsets of wallets and
+dapps on a given chain, which requires semantic negotiation and/or network
+routing negotiation before authorization can occur.
+
+## Specification
+
+Two properties are added to the scope objects requested in `optionalScopes` and
+optionally returned in `sessionScopes`.  These are both [ordered] strings of arrays:
+
+1. `rpcEndpoints` is an array of zero or more URLs of RPC endpoints that
+   the caller would prefer the respondent to use, ordered by preference. Each
+   must be a valid URL that addresses an RPC endpoint. The respondent may
+   return it empty, reordered, with less, the same, or even more conformant URLs
+   than received.
+2. `rpcDocuments` is an array of zero or more URLs of machine-readable RPC
```

Comment:
What's the expected behavior if zero `rpcEndpoints` and zero `rpcDocuments` are provided?

Reply:
I threw together a straw man in a new commit but we should probably iterate this a bit, [it's kinda clunky as-is](https://github.com/ChainAgnostic/CAIPs/pull/211/commits/8703b014734361eccf3bb41cfbd9283b641f6f3a)